### DISCUSSION TITLE
"X-JWT-Assertion" security scheme

### DISCUSSION THREAD

**Discussion by Angel0926** [quarkusio/quarkus#36156] [Community]
How i can realize my own security-scheme: jwt "X-JWT-Assertion" ?

**Comment by sberyozkin** [comment]
@Angel0926 Do you mean that you have the bearer access token set with HTTP `X-JWT-Assertion` header, `X-JWT-Assertion <token>`, instead of the default `Authorization: Bearer <token>` ? If yes, use `mp.jwt.token.header=X-JWT-assertion`, see https://quarkus.io/guides/security-jwt#microprofile-jwt-configuration

**Comment by Angel0926** [comment]
Yes my project in quarkus and also i added this code in my application.yaml `    

    security-scheme: jwt
    jwt-security-scheme-value: "bearer"
    jwt-bearer-format: "JWT"
    security-scheme-name: "X-JWT-Assertion"
    security-scheme-description: "X-JWT-Assertion"
    auto-add-security-requirement: true`.    what change ?

**Comment by Angel0926** [comment]
When i try portman newman integration i get this 
<img width="382" alt="Screenshot 2023-09-26 at 14 18 03" src="https://github.com/quarkusio/quarkus/assets/91777306/2cd6e145-f202-42bf-bb15-7aa26f0279c7">

**Comment by sberyozkin** [comment]
@Angel0926 I don't recognize those properties, which extension do you use ?

**Comment by Angel0926** [comment]
@sberyozkin i find it here https://quarkus.io/guides/openapi-swaggerui

**Comment by sberyozkin** [comment]
@Angel0926 I see, I guess those properties can help to show or possibly test from within Swagger UI, they don't, in itself control any verification process, to verify the token coming with the custom header, you'd need to use `quarkus-smallrye-jwt`, or `quarkus-oidc`

**Comment by Angel0926** [comment]
@sberyozkin   @SecurityScheme(securitySchemeName = “jwt”, type = SecuritySchemeType.HTTP, scheme = “bearer”, bearerFormat = “jwt”) This code right? and what it doing?

**Comment by sberyozkin** [comment]
@Angel0926 AFAIK, this code is about telling SwaggerUI/OpenAPI how to show and possibly test this method with Swagger UI. See my previous comment about the verification
